### PR TITLE
Sanitize svelte @html bindings to prevent XSS

### PR DESCRIPTION
**Pull request by ozsivanov**
# Code changes

- Sanitize `@html` tags with DOMPurify
- Removed `@html` tags in `NError.svelte`

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: commons/src/components/NError.svelte [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
-  $: error = $ErrorStore[id] || { name: "" };
-  $: explanation = NErrorExplanationMap[errorName]();
-  $: errorName = !error.name ? error.message?.name : error.name;
-  const isDevEnv = window.location.href.includes("localhost");
+  let error: NError;
+  let errorName: string;
+
+  $: {
+    error = $ErrorStore[id] ?? { name: "" };
+    errorName = error.name ?? error.message?.name ?? "";
```
Comment (heathergh) [inline commons/src/components/NError.svelte:14]: double nullish coalescing?! 🤯

File: components/agenda/src/Agenda.svelte [hunk 2]
```diff
@@ -1426,7 +1427,7 @@
                       </span>
                     {/if}
                     <p>
-                      {@html event.description || ""}
+                      {@html DOMPurify.sanitize(event?.description ?? "")}
```
Comment (heathergh) [inline components/agenda/src/Agenda.svelte:1430]: Is this optional chaining required for `event`? This element is inside of an `{#each allDayEvents as event}`

File: components/agenda/src/Agenda.svelte [hunk 3]
```diff
@@ -1530,7 +1526,7 @@
...
```
Comment (heathergh) [inline components/agenda/src/Agenda.svelte:1529]: same q here for `event?`

File: components/email/src/init.spec.js [hunk 4]
```diff
@@ -276,6 +276,27 @@ describe("Email component", () => {
...
+          ...SAMPLE_THREAD.messages[0],
+          body: `<img src=x onerror=alert(1)//>`,
+        };
+        setTimeout(() => {
+          expect(stub).not.to.be.called;
+          done();
+        }, 50);
+      });
+  });
+
```
Comment (heathergh) [inline components/email/src/init.spec.js:299]: Yay! A test!

**Review by heathergh — APPROVED** [review]
LGTM!

**Review by philrenaud — APPROVED** [review]

File: components/agenda/src/Agenda.svelte (hunk 2 above)
Comment (ozsivanov) [reply components/agenda/src/Agenda.svelte:1430]: oh yeah, good catch. It isn't needed.

**Pull request merged**